cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/connector/ConnectRequest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.etl.api.connector;
+
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Connect request for a connector
+ */
+public class ConnectRequest {
+
+  /**
+   * The option that is possible for the request
+   */
+  public enum ConnectorOption {
+    ALL,
+    LIMIT,
+    PROPERTIES
+  }
+
+  protected final String path;
+  protected final Map<ConnectorOption, ?> options;
```

Comment:
What do we store in the Map? Why it is a Map to anything? A class like this will make it cannot be serialized.

Reply:
I'm also not clear on what these options are, but it seems like they should be separate optional fields. A field for the limit, a field for properties. (What are example of properties?)